android/app/src/main/java/com/created/team201/presentation/studyList/StudyListFragment.kt

```diff
@@ -239,28 +247,28 @@ class StudyListFragment : BindingFragment<FragmentStudyListBinding>(R.layout.fra
 
     private fun setupStudyListFilter() {
         binding.cgStudyList.setOnCheckedStateChangeListener { group, _ ->
-            when (group.checkedChipId) {
-                binding.chipStudyListAll.id -> {
-                    studyListViewModel.loadFilteredPage(StudyListFilter.ALL)
-                }
-
-                binding.chipStudyListWaiting.id -> {
-                    studyListViewModel.loadAppliedPage(mainViewModel.isGuest)
-                }
-
-                binding.chipStudyListProcessing.id -> {
-                    studyListViewModel.loadFilteredPage(StudyListFilter.PROCESSING)
-                }
+            loadFilteredPage(group)
+        }
+    }
 
-                binding.chipStudyListRecruiting.id -> {
-                    studyListViewModel.loadFilteredPage(StudyListFilter.RECRUITING)
-                }
+    private fun loadFilteredPage(group: ChipGroup) {
+        studyListViewModel.updateIsGuest(mainViewModel.isGuest)
+        val filter = getStudyListFilter(group)
+        studyListViewModel.loadFilteredPage(filter)
+    }
 
-                -1 -> {
-                    group.check(R.id.chip_study_list_all)
-                    studyListViewModel.loadFilteredPage(StudyListFilter.ALL)
-                }
+    private fun getStudyListFilter(group: ChipGroup): StudyListFilter {
+        return when (group.checkedChipId) {
+            binding.chipStudyListAll.id -> StudyListFilter.ALL
+            binding.chipStudyListWaiting.id -> StudyListFilter.WAITING
+            binding.chipStudyListProcessing.id -> StudyListFilter.PROCESSING
+            binding.chipStudyListRecruiting.id -> StudyListFilter.RECRUITING
+            NON_SELECTED_CHIP_ID -> {
+                group.check(R.id.chip_study_list_all)
+                StudyListFilter.ALL
             }
+
+            else -> throw IllegalArgumentException()
```

Comment:
Filter enum class의 companion object에서 처리하는건 어떨거 같나요?

Reply:
이게 chip 아이디를 연결시켜주는거라... enum class 안에서는 처리를 못할 것 같습니다. (다른 아이디어 있으면 말씀해주세요)

```suggestion
    private fun getStudyListFilter(group: ChipGroup): StudyListFilter {
        return when (group.checkedChipId) {
            NON_SELECTED_CHIP_ID -> {
                group.check(R.id.chip_study_list_all)
                StudyListFilter.ALL
            }
            else -> {
                val chip: CustomChip = group.findViewById(group.checkedChipId)
                return chip.filter
            }
        }
        }
```

아니면 아래와 같이 하는 방법도 있을 것 같습니당.
변경될 가독성은 떨어지지만 (else가 chip이 선택된 경우) 코드는 더 간결해지네요 !

Reply:
반달 의견 들어보니 아래거가 더 낫다고 해서 반영했습니다 !